Python/compile.c

```diff
@@ -124,16 +126,24 @@ is_relative_jump(struct instr *i)
     return is_bit_set_in_table(_PyOpcode_RelativeJump, i->i_opcode);
 }
 
+static inline int
+is_block_push(struct instr *instr)
+{
+    int opcode = instr->i_opcode;
+    return opcode == SETUP_FINALLY || opcode == SETUP_WITH || opcode == SETUP_CLEANUP;
+}
+
 static inline int
 is_jump(struct instr *i)
 {
-    return i->i_opcode >= SETUP_WITH || is_bit_set_in_table(_PyOpcode_Jump, i->i_opcode);
+    return is_bit_set_in_table(_PyOpcode_Jump, i->i_opcode);
```

Comment:
Ah wait, I think calling `is_bit_set_in_table` on virtual opcodes is illegal. Probably still need a guard here?

```suggestion
    return !IS_VIRTUAL_OPCODE(i->i_opcode) && is_bit_set_in_table(_PyOpcode_Jump, i->i_opcode);
```